dataset/bin.cpp

```diff
@@ -263,16 +263,24 @@ auto axis_actions(const VariableConstView &var,
   // also need to handle bin contents from all dimensions inside the rebinned
   // one, even if the grouping/binning along this dimension is unchanged.
   bool rebin = false;
-  const auto dims = var.dims();
+  const auto dims = array.dims();
   for (const auto dim : dims.labels()) {
     if (edges_dims.contains(dim) || groups_dims.contains(dim))
       rebin = true;
     if (groups_dims.contains(dim))
       axes.emplace_back(AxisAction::Group, dim, groups[groups_dims.index(dim)]);
     else if (edges_dims.contains(dim))
       axes.emplace_back(AxisAction::Bin, dim, edges[edges_dims.index(dim)]);
-    else if (rebin)
+    else if (rebin) {
+      if (array.coords().contains(dim) &&
+          array.coords()[dim].dims().ndim() != 1)
+        throw except::DimensionError(
+            "2-D coordinate " + to_string(array.coords()[dim]) +
+            " conflicting with (re)bin of outer dimension. Try specifying new "
```

Comment:
Given that this message will propagate to user. Could you be more explicit? `i.e run bin on dimension {dimension}`

Reply:
See update.